### PR TITLE
Stats: Updating details page links placement

### LINKED ISSUE #70849: Stats: Adjust link styling for clickable headers
- [ ] Apply the animated link indicator on the annual highlights section on the Insights page.

Image

- [ ] Apply the animated link indicator on the stats modules on the Traffic page.

Image

### PR DESCRIPTION
**Pull request by grzegorz-cp**
#### Proposed Changes

* adding an additional text for the highlight section on the `Insights` page, making the header link more visible
* removing the header link from the components with horizontal bars and moving it to the bottom of the card to unify the navigation

This PR is reflecting conversations from [#70905](https://github.com/Automattic/wp-calypso/pull/70905#issuecomment-1341770162)

#### Testing Instructions

* use the live branch to get to the Stats page
* make sure that all of the components having horizontal bars (in the `Traffic` page) don't have links on the component's headers
* make sure that all of the components with horizontal bars have a link at the bottom of the card, navigating to its details page
* components with 10 items should have a `View all` link, while components with fewer items should have a `View details` link
* highlight components on the `Insights` page should have a `View all annual insights` link next to their title
* all of the components on the `Insights` page shouldn't be affected by the title and bottom link changes made to the `Traffic` page

Before:
Screenshot 2022-12-09 at 10 32 51 AM
Screenshot 2022-12-09 at 10 33 47 AM

After:
Screenshot 2022-12-09 at 10 35 40 AM
Screenshot 2022-12-09 at 10 34 21 AM

#### Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Fixes #70849.

### CODE REVIEW

File: client/my-sites/stats/stats-module/index.jsx [hunk 1]
```diff
@@ -222,7 +226,7 @@ class StatsModule extends Component {
 								<StatsModulePlaceholder isLoading={ isLoading } />
 								<StatsList moduleName={ path } data={ data } useShortLabel={ useShortLabel } />
 							</div>
-							{ this.props.showSummaryLink && displaySummaryLink && (
+							{ this.props.showSummaryLink && data && data.length >= 10 && (
```
Comment (jsnmoon) [inline client/my-sites/stats/stats-module/index.jsx]:
```suggestion
							{ this.props.showSummaryLink && data?.length >= 10 && (
```

⛳

**Review by jsnmoon — CHANGES_REQUESTED** [review]
I recommend two changes, but I'll defer to our designers' opinions.

image

- [x] The small text for 7-day highlights isn't a link and shouldn't look like one.

image

- [x] The header for annual highlights should no longer be a link. Only the small text for the header should be a link.

**Comment by jeffgolenski** [comment]
Brilliant @grzegorz-cp! Overall, it all works great. Only 3 items to address:

* The two items @jsnmoon mentioned above
* On hover / active can we have an underline for any "view all" and "view details" link?

Overall, I think @keoshi and myself need to run an audit for consistency within hover and active states everywhere. But for now, this is great! Thanks again!

**Comment by grzegorz-cp** [comment]
Thanks team, it's ready for another review.

File: client/my-sites/stats/stats-countries/index.jsx [hunk 2]
```diff
@@ -25,6 +25,7 @@ class StatCountries extends Component {
 				path="countryviews"
 				period={ period }
 				showSummaryLink={ ! summary }
+				hideSummaryLink={ !! summary }
 				summary={ summary }
```
Comment (kangzj) [inline client/my-sites/stats/stats-countries/index.jsx:29]: The `summary` is a bit confusing here. Any ways to make it clearer?

File: client/my-sites/stats/stats-module/index.jsx [hunk 3]
```diff
@@ -172,16 +172,20 @@ class StatsModule extends Component {
...
 						emptyMessage={ moduleStrings.empty }
 						showMore={
-							this.props.showSummaryLink && displaySummaryLink
+							displaySummaryLink
 								? {
 										url: this.getHref(),
-										label: this.props.translate( 'View all', {
-											context: 'Stats: Button label to expand a panel',
-										} ),
+										label:
```
Comment (kangzj) [inline client/my-sites/stats/stats-module/index.jsx:176]: I understand this is by design - it's just a little strange that `View all` doesn't really expand the panel.

**Review by kangzj — APPROVED** [review]
It works as described - a couple of minor comments no blockers.

File: client/my-sites/stats/stats-module/index.jsx (hunk 3 above)
Comment (jsnmoon) [reply client/my-sites/stats/stats-module/index.jsx:176]: Agreed, the translator's comment doesn't accurately portray what the link does. Perhaps it should be more along the lines of "Button link to show more detailed stats information".

**Review by jsnmoon — APPROVED** [review]
I added an underline for the subheader link in the Annual Highlights section. (I also removed new styles that were being overridden by existing rules with higher specificity.)

This PR looks good to ship 👍

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7671316

Thank you @grzegorz-cp for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7671316).

**Pull request merged**